extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -709,16 +709,30 @@ export class Bind {
         break;
 
       default:
+        // Some input elements treat some of their attributes as initial values.
+        // Once the user interacts with these elements, the JS properties
+        // underlying these attributes must be updated for the change to be
+        // visible to the user.
+        const updateElementProperty =
+            element.tagName == 'INPUT' && property in element;
         const oldValue = element.getAttribute(property);
 
         let attributeChanged = false;
         if (typeof newValue === 'boolean') {
-          if (newValue && oldValue !== '') {
-            element.setAttribute(property, '');
+          if (updateElementProperty && element[property] !== newValue) {
+            // Property value *must* be read before the attribute is changed.
+            // Before user interaction, attribute updates affect the property.
+            element[property] = newValue;
             attributeChanged = true;
-          } else if (!newValue && oldValue !== null) {
+          } else {
+            attributeChanged =
+                (oldValue === null && newValue) ||
+                (oldValue === '' && !newValue);
```

Comment:
Still think the old code was easier to read. 😄  Also you'd avoid useless `setAttribute`/`removeAttribute` calls on the element.